public/app/features/dashboard/components/DashNav/DashNav.tsx

```diff
@@ -274,6 +275,15 @@ export const DashNav = React.memo<Props>((props) => {
   const parentHref = locationUtil.getUrlForPartial(location, { search: 'open', folder: 'current' });
   const onGoBack = isFullscreen ? onClose : undefined;
 
+  if (config.featureToggles.topnav) {
+    return (
+      <TopNavUpdate
+        subNav={{ text: title }}
+        actions={<ToolbarButton onClick={onOpenSettings} icon="cog"></ToolbarButton>}
+      />
+    );
+  }
```

Comment:
Nice! This was more or less what I had in mind! (though, maybe i think we should set the pattern of only doing this at the top of the page, rather than down in subcomponents maybe?)

I think putting actions in JSX children (`<TopNavAction><ToolbarButtons ... /></TopNavAction>` is a bit nicer personally, but just a minor nit.